sources/httpUtils.ts

```diff
@@ -14,11 +14,11 @@ export async function fetchUrlStream(url: string, options: RequestOptions = {})
 
   return new Promise<IncomingMessage>((resolve, reject) => {
     const request = https.get(url, {...options, agent: proxyAgent}, response => {
-      const statusCode = response.statusCode ?? 500;
-      if (!(statusCode >= 200 && statusCode < 300))
-        return reject(new Error(`Server answered with HTTP ${statusCode}`));
+      const statusCode = response.statusCode;
+      if (statusCode && statusCode >= 200 && statusCode < 300)
```

Comment:
We want an error if the status code is 0
```suggestion
      if (statusCode != null && statusCode >= 200 && statusCode < 300)
```

Reply:
I'm afraid that typecheck complains with "TS2532: object is possibly 'undefined'."

Reply:
`!= null` catches both `null` and `undefined`, Ts shouldn’t complain here

Reply:
Cool!

Reply:
Updated.